### ISSUE TITLE
Conflict with CleanURL plugin

### ISSUE THREAD

**Issue by hxsllc** [hafizchin/DomainManager#2]
Wondering if you would be able to address this.

https://github.com/biblibre/omeka-s-module-CleanUrl

When OmekaSDomainManager plugin is enabled, it interferes with CleanUrl plugin...

Result of CleanUrl plugin is the following URL, almost all plugin settings set to default:
http://edisondigital.rutgers.edu/s/digital/document/Z400CK

Prior to CleanURL plugin being activated, the URL looks like:
http://edisondigital.rutgers.edu/s/digital/item/154520

With OmekaSDomainManager plugin being activated, the URL looks like:
http://edisondigital.rutgers.edu/item/154520

Trying to reactivate CleanURL plugin after activated OmekaSDomainManager, the URL /document/Z400CK results in:
http://edisondigital.rutgers.edu/document/Z400CK > "Omeka S encountered an error"
![Screen Shot 2019-06-20 at 9 51 39 AM](https://user-images.githubusercontent.com/29841283/59854489-09ea0600-9341-11e9-9098-48e74f8319da.png)

**Comment by hafizchin** [comment]
Hi,

Have you tried using the CleanURL plugin while  the OmekaSDomainManager plugin is disabled or deactivated?

**Comment by hxsllc** [comment]
Yes, correct, sorry that wasn't clear. It's verified that this plugin interferes (I assume) with CleanUrl functioning properly.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by hafizchin** [comment]
Hi,

Looks like the issue is in the CleanURL plugin.

![image](https://user-images.githubusercontent.com/6478018/59865142-52f88500-9356-11e9-83ed-f47ae38c6998.png)

If you look at the screenshot, the url segment "document" is interpreted by omeka-s as a Controller and that controller does not exist.

**Comment by hxsllc** [comment]
Ok I will ask Daniel about that, thank you!

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by hxsllc** [comment]
I appreciate the quick response! And I've asked the developer to respond. However, I don't think this analysis is correct. The plugin takes a setting as the route/controller, per attached screenshots. So the plugin handles a user-defined "path" for the "friendly URL" version of the item path.
![Screen Shot 2019-06-20 at 1 59 24 PM](https://user-images.githubusercontent.com/29841283/59870573-c6ed5a00-9363-11e9-9b34-d525de98fa76.png)
![Screen Shot 2019-06-20 at 1 59 05 PM](https://user-images.githubusercontent.com/29841283/59870579-c94fb400-9363-11e9-9ee4-53f14502b234.png)

**Comment by hafizchin** [comment]
As per your previous reply, you said you've tried using the CleanURL while the OmekaSDomainManager is disabled. While doing this, did you encounter the same error?

**Comment by Daniel-KM** [comment]
Yes, they work fine separately, not together. I don't know yet in which module it's easier to fix it.

**Comment by hxsllc** [comment]
Well more specifically, CleanUrl does not stop DomainManager from functioning properly in terms of how it rewrites the URL / path, but DomainManager appears to stop the rewrite functionality of CleanUrl. Maybe it's a case of only one PHP class being able to "own" modifying paths? I assume both modules are using the same method of modifying URL.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by hafizchin** [comment]
Hi Daniel,

I have deployed a fix, do let me know if this solves your previous issue.

Regards,

Mo

**Comment by hxsllc** [comment]
Awesome! This is great, I will test it out right away.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by hxsllc** [comment]
I tried it out but got the same result - an error page. I didn't have time to look at the PHP logs but can do so soon.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by hafizchin** [comment]
Can you give me the url? So i can test.

Thanks.

**Comment by hafizchin** [comment]
Hi,

I've deployed a fix, can you give it a try and let me know if it resolves your issue.

Regards,
Mo

**Comment by hxsllc** [comment]
When I tried now, I got a different result - the Browse Items page wouldn't load any items, but did not produce an error.

I have a bunch of screenshots I can send your way if you want - email me at benbakelaar@gmail.com.

The site (without DomainManager activated right now) is http://edisondigital.rutgers.edu. You'll see it adds in the /s/digital as the primary site. And you'll see as in the original post that if you go to a document, instead of /item/internal_id, the URL for the item will be /document/dcterms_identifier (per configuration in CleanURL plugin).
